apps/web/src/routes/dashboard/_dashboard/-components/booking-table.tsx

```diff
@@ -0,0 +1,185 @@
+}
+
+const convertToCSV = (data: Participant[]) => {
+  const header = ['Index', 'Username', 'Display Name'].join(',');
+
+  const rows = data.map((row, index) =>
+    [index + 1, row.username, row.displayName || '']
+      .map((value) => `"${value}"`)
+      .join(','),
+  );
+
+  return [header, ...rows].join('\n');
+};
+
+const handleDownload = (participants: Participant[], eventName: string) => {
+  if (!participants || participants.length === 0) {
+    console.error('No participants available for download.');
+    return;
+  }
+
+  const csvData = convertToCSV(participants);
+
+  const blob = new Blob([csvData], { type: 'text/csv' });
+  const link = document.createElement('a');
+  link.href = URL.createObjectURL(blob);
+
+  const sanitizedEventName = eventName
+    .replaceAll(/[^\da-z]/gi, '_')
+    .toLowerCase();
+  const filename = `${sanitizedEventName}_participants.csv`;
+  link.download = filename;
+  link.click();
+};
+
+const BookingTable = () => {
+  const { t } = useTranslation();
+
+  const { data: allEvents, isFetched } =
+    trpc.user.calendar.getCalendarEvents.useQuery({
+      language: 'en',
+      upcomingEvents: true,
+    });
+
+  return (
+    <div>
+      {!isFetched && <Loader size={'s'} />}
+
+      <div className="overflow-x-auto max-md:hidden md:flex">
+        <Table className="!max-w-[910px]">
+          <TableHeader className="border-b-0">
+            <TableRow>
+              <TableHead className="w-[137px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.type')}
+              </TableHead>
+              <TableHead className="w-[420px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.nameOfEvent')}
+              </TableHead>
+              <TableHead className="w-[180px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.date')}
+              </TableHead>
+              <TableHead className="flex justify-center w-[180px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.listOfParticipants')}
+              </TableHead>
+            </TableRow>
+          </TableHeader>
+          <TableBody>
+            {allEvents?.map((event) => (
+              <EventRow key={event.id} event={event} />
+            ))}
+          </TableBody>
+        </Table>
+      </div>
+
+      <div className="flex flex-col gap-4 md:hidden">
+        {allEvents?.map((event) => <EventCard key={event.id} event={event} />)}
+      </div>
+    </div>
+  );
+};
+
+interface EventProps {
+  event: Event;
+}
+
+const EventRow = ({ event }: EventProps) => {
+  const { t } = useTranslation();
+
+  const { data: participants, isFetched } =
+    trpc.user.events.getParticipantsForEvent.useQuery({
+      eventId: event.id,
+    });
+
+  const startDate = event.startDate || new Date();
+
+  return (
+    <TableRow>
+      {!isFetched && <Loader size={'s'} />}
```

Comment:
Should be outside (incorrect html)

![image](https://github.com/user-attachments/assets/b1975c4f-c32f-4127-9f23-8c0a3d6787ad)